src/class-otgs-remote-file-cache.php

```diff
@@ -2,13 +2,21 @@
 
 class OTGS_Remote_File_Cache {
 
-	const OPTION_KEY_PREFIX = '_otgs_remote_file_hash_';
+	const OPTION_KEY_PREFIX = 'otgs_remote_file_hash';
 
 	private $file_url;
+	private $option_key_name;
 	private $remote_hash;
 
-	public function __construct( $file_url ) {
+	/**
+	 * OTGS_Remote_File_Cache constructor.
+	 *
+	 * @param string $file_url
+	 * @param string $option_key_name
+	 */
+	public function __construct( $file_url, $option_key_name ) {
```

Comment:
@grigorij add some sanitization here, $option_key_name_should be string in the end. It works like an API, if user will pass an array as $option_key_name it will break with fatal error. For the file_url it is fine because I used md5.

Reply:
@grigorij after our discussion. Let's add two exceptions here if one of the constructor values are invalid (not string). So the caller will know how to fix it instead of seeing fatal error when using `sprintf `.

Reply:
added